src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpMtlsHttpClientFactory.cs

```diff
@@ -0,0 +1,137 @@
+                }
+            }
+
+            if (!string.IsNullOrEmpty(mtlsOptions.ClientCertificatePath))
+            {
+                if (string.IsNullOrEmpty(mtlsOptions.ClientKeyPath))
+                {
+                    // Check if certificate file exists to provide appropriate error message
+                    if (!File.Exists(mtlsOptions.ClientCertificatePath))
+                    {
+                        throw new FileNotFoundException($"Certificate file not found at path: {mtlsOptions.ClientCertificatePath}", mtlsOptions.ClientCertificatePath);
+                    }
+                }
+                else
+                {
+                    clientCertificate = OtlpMtlsCertificateManager.LoadClientCertificate(
+                        mtlsOptions.ClientCertificatePath,
+                        mtlsOptions.ClientKeyPath,
+                        mtlsOptions.ClientKeyPassword);
+
+                    if (mtlsOptions.EnableCertificateChainValidation)
+                    {
+                        OtlpMtlsCertificateManager.ValidateCertificateChain(
+                            clientCertificate,
+                            OtlpMtlsCertificateManager.ClientCertificateType);
+                    }
+
+                    OpenTelemetryProtocolExporterEventSource.Log.MtlsConfigurationEnabled(
+                        clientCertificate.Subject);
+                }
+            }
+
+            // Create HttpClientHandler with mTLS configuration
+#pragma warning disable CA2000 // Dispose objects before losing scope - HttpClientHandler is disposed by HttpClient
+            handler = new HttpClientHandler { CheckCertificateRevocationList = true };
+#pragma warning restore CA2000
+
+            // Add client certificate if available
+            if (clientCertificate != null)
+            {
+                handler.ClientCertificates.Add(clientCertificate);
+                handler.ClientCertificateOptions = ClientCertificateOption.Manual;
+            }
+
+            // Set up server certificate validation
+            if (caCertificate != null)
+            {
+                handler.ServerCertificateCustomValidationCallback = (
+                    httpRequestMessage,
+                    cert,
+                    chain,
+                    sslPolicyErrors) =>
+                {
+                    if (cert == null || chain == null)
+                    {
+                        return false;
+                    }
+
+                    return OtlpMtlsCertificateManager.ValidateServerCertificate(
+                        cert, chain, sslPolicyErrors, caCertificate);
+                };
+            }
+
+            var client = new HttpClient(handler, disposeHandler: true);
+
+            configureClient?.Invoke(client);
+
+            // Dispose certificates as they are no longer needed after being added to the handler
+            caCertificate?.Dispose();
+            clientCertificate?.Dispose();
```

Comment:
This should be in a `try-catch` so that if these throw then we still dispose of `client` as it won't be returned to the caller.